source/fuzz/fuzzer_pass_construct_composites.cpp

```diff
@@ -345,18 +330,16 @@ FuzzerPassConstructComposites::FindComponentsToConstructVector(
                                          available_instructions->second.begin(),
                                          available_instructions->second.end());
     }
-    if (instructions_to_choose_from.empty()) {
-      // We may get unlucky and find that there are not any instructions to
-      // choose from.  In this case we give up constructing a composite of this
-      // vector type.  It might be that we could construct the composite in
-      // another manner, so we could opt to retry a few times here, but it is
-      // simpler to just give up on the basis that this will not happen
-      // frequently.
-      return {};
-    }
-    auto instruction_to_use =
-        instructions_to_choose_from[GetFuzzerContext()->RandomIndex(
-            instructions_to_choose_from)];
+    // If there are no instructions to choose from then use a zero constant,
+    // otherwise select one of the instructions at random.
+    uint32_t id_of_instruction_to_use =
+        instructions_to_choose_from.empty()
+            ? FindOrCreateZeroConstant(element_type_id,
+                                       GetFuzzerContext()->ChooseEven())
+            : instructions_to_choose_from[GetFuzzerContext()->RandomIndex(
+                  instructions_to_choose_from)];
+    opt::Instruction* instruction_to_use =
+        GetIRContext()->get_def_use_mgr()->GetDef(id_of_instruction_to_use);
```

Comment:
The comment regarding lines https://github.com/KhronosGroup/SPIRV-Tools/blob/75de8be2425de24ad9a50f61cce7eb24bc3ff9ad/source/fuzz/fuzzer_pass_construct_composites.cpp#L360-L365 - we can replace the while loop with a call to `FuzzerContext::Shuffle` which is linear instead of quadratic.

It also seems that `std::vector<uint32_t> operands` is not used anywhere (don't know why gcc doesn't produce an error).

Reply:
Done. And yes, weird about gcc not complaining.